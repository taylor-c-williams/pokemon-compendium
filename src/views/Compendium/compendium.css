@import url('https://fonts.googleapis.com/css2?family=DM+Mono:ital,wght@0,300;1,500&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Modak&display=swap');

html,
body {
	font-family: 'DM Mono', monospace;
	letter-spacing: 0.25em;
	margin: 50px;
	margin-top: 0px;
}

.loading {
	font-size: 4em;
	font-family: 'DM Mono', monospace;
	color: black;
	display: flex;
	justify-content: center;
	align-items: center;
	text-align: center;
	background-color: rgba(255, 255, 255, 0.9);
	border-radius: 100px;
	padding: 30px;
	-webkit-animation-name: hvr-hang-sink, hvr-hang;
	animation-name: hvr-hang-sink, hvr-hang;
	-webkit-animation-duration: 0.3s, 1.5s;
	animation-duration: 0.3s, 1.5s;
	-webkit-animation-delay: 0s, 0.3s;
	animation-delay: 0s, 0.3s;
	-webkit-animation-timing-function: ease-out, ease-in-out;
	animation-timing-function: ease-out, ease-in-out;
	-webkit-animation-iteration-count: 1, infinite;
	animation-iteration-count: 1, infinite;
	-webkit-animation-fill-mode: forwards;
	animation-fill-mode: forwards;
	-webkit-animation-direction: normal, alternate;
	animation-direction: normal, alternate;
}

h1 {
	font-size: 9em;
	font-family: 'Modak', cursive;
	color: white;
	font-weight: 100;
	text-shadow: -5px -5px rgba(226, 190, 47, 1);
	text-align: center;
}

h2 {
	font-size: 1.5em;
	margin: 0;
	text-transform: capitalize;
	text-align: center;
	letter-spacing: 0.35em;
}

h3 {
	margin: 0;
	font-size: 1.1em;
	letter-spacing: 0.35em;
}

select {
	margin: 35px;
	margin-top: 5px;
	width: 200px;
	height: auto;
	font-size: 1.5em;
	font-family: 'DM Mono', monospace;
	text-transform: capitalize;
	border: none;
}

select:focus {
	outline: none;
}

.compendium-body {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	margin: 25px;
}

@-webkit-keyframes hvr-hang {
	0% {
		-webkit-transform: translateY(8px);
		transform: translateY(8px);
	}
	50% {
		-webkit-transform: translateY(4px);
		transform: translateY(4px);
	}
	100% {
		-webkit-transform: translateY(8px);
		transform: translateY(8px);
	}
}
@keyframes hvr-hang {
	0% {
		-webkit-transform: translateY(8px);
		transform: translateY(8px);
	}
	50% {
		-webkit-transform: translateY(4px);
		transform: translateY(4px);
	}
	100% {
		-webkit-transform: translateY(8px);
		transform: translateY(8px);
	}
}
@-webkit-keyframes hvr-hang-sink {
	100% {
		-webkit-transform: translateY(8px);
		transform: translateY(8px);
	}
}
@keyframes hvr-hang-sink {
	100% {
		-webkit-transform: translateY(8px);
		transform: translateY(8px);
	}
}
