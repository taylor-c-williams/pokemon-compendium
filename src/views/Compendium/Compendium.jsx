import { useState, useEffect } from 'react';
import {
	fetchPokemon,
	fetchTypes,
	fetchFilteredPokemon,
} from '../../services/utils';
import PokeList from '../../components/PokeList/PokeList';
import Controls from '../../components/Controls/Controls';
import './compendium.css';

export default function Compendium() {
	// Set State
	const [pokedex, setPokedex] = useState([]);
	const [loading, setLoading] = useState(true);
	const [allTypes, setAllTypes] = useState([]);
	const [selectedType, setSelectedType] = useState('');
	const [sortOrder, setSortOrder] = useState('asc');

	// Get Pokemon, Get Types
	useEffect(() => {
		async function getPokemon() {
			const pokemonList = await fetchPokemon();
			setPokedex(pokemonList);
			setLoading(false);
		}
		async function getAllTypes() {
			const types = await fetchTypes();
			setAllTypes(types);
		}
		getPokemon();
		getAllTypes();
	}, []);

	// Filter by Type & Set Sort Order
	useEffect(() => {
		async function getFilteredPokemon() {
			setLoading(true);
			if (!selectedType || selectedType === 'all') {
				const pokemonList = await fetchPokemon(sortOrder);
				setPokedex(pokemonList);
			} else {
				const filteredPokemon = await fetchFilteredPokemon(
					selectedType,
					sortOrder
				);
				setPokedex(filteredPokemon);
			}
			setLoading(false);
		}
		getFilteredPokemon();
	}, [selectedType, sortOrder]);

	return (
		<div className="compendium-body">
			{/* <h1>Pokemon Compendium</h1> */}
			<section className="header">
				<h1>Pokemon Compendium</h1>
			</section>
			{loading ? (
				<section className="loading">
					<h2> Loading ... </h2>
				</section>
			) : (
				<section className="components">
					<Controls
						allTypes={allTypes}
						selectedType={selectedType}
						setSelectedType={setSelectedType}
						sortOrder={sortOrder}
						setSortOrder={setSortOrder}
					/>

					<PokeList pokedex={pokedex} selectedType={selectedType} />
				</section>
			)}
		</div>
	);
}
