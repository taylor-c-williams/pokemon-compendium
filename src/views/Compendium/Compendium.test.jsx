import { screen, render } from '@testing-library/react';
import Compendium from './Compendium';

it('accurately renders the Compendium view', () => {
	render(<Compendium />);
	const loading = screen.getAllByText('Loading ...');
	expect(loading).toMatchSnapshot();
});
