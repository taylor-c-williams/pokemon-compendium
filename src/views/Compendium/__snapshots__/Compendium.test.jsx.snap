// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accurately renders the Compendium view 1`] = `
Array [
  <h2>
     Loading ... 
  </h2>,
]
`;
